src/utils/child_process.rs

```diff
@@ -89,8 +91,135 @@ impl Nanny {
 }
 
 fn boot_desktop_file(path: &Path) -> std::io::Result<Child> {
-    let args = format!( "`if [ \"$(grep '^X-GNOME-Autostart-enabled' {:?} | tail -1 | sed 's/^X-GNOME-Autostart-enabled=//' | tr '[A-Z]' '[a-z]')\" != 'false' ]; then grep '^Exec' {:?} | tail -1 | sed 's/^Exec=//' | sed 's/%.//' | sed 's/^\"//g' | sed 's/\" *$//g'; else echo 'exit'; fi`", path , path);
-    Command::new("sh").arg("-c").arg(args).spawn()
+    use std::io::Error as StdError;
+    let entry = DesktopEntry::parse_file(path)?;
+    let dw_env = std::env::var("XDG_CURRENT_DESKTOP").unwrap_or(Default::default());
+
+    if entry.only_show_in.is_some() {
+        if !entry.only_show_in.unwrap().contains(&dw_env) {
+            return Err(StdError::from(std::io::ErrorKind::Other));
+        }
+    }
+    if entry.not_show_in.is_some() {
+        if entry.not_show_in.unwrap().contains(&dw_env) {
+            return Err(StdError::from(std::io::ErrorKind::Other));
+        }
+    }
+    if entry.hidden {
+        return Err(StdError::from(std::io::ErrorKind::Other));
+    }
+    if entry.exec.is_none() {
+        return Err(StdError::from(std::io::ErrorKind::Other));
+    }
+    let wd = entry
+        .path
+        .unwrap_or(dirs_next::home_dir().unwrap_or(PathBuf::from(".")));
+
+    Command::new("sh")
+        .current_dir(wd)
+        .arg("-c")
+        .arg(entry.exec.unwrap())
+        .spawn()
+}
+
+/// # DesktopEntry
+/// Just need to parse some keys that are listed in [Recognized desktop entry keys](https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s06.html).
+#[derive(Debug, Default)]
+struct DesktopEntry {
+    // TryExec: Option<String>,
+    exec: Option<String>,
+    path: Option<PathBuf>,
+    only_show_in: Option<HashSet<String>>,
+    not_show_in: Option<HashSet<String>>,
+    hidden: bool,
+}
+
+impl DesktopEntry {
+    fn parse_file(path: &Path) -> std::io::Result<Self> {
+        let content = fs::read_to_string(path)?;
+        return Self::parse(content.as_str());
+    }
+    fn parse(content: &str) -> std::io::Result<Self> {
+        let mut in_main_section = false;
+        let mut entry: Self = Default::default();
+        for mut line in content.lines() {
+            line = line.trim();
+
+            if line.is_empty() || line.starts_with('#') {
+                continue;
+            }
+
+            if line.starts_with('[') {
+                if line == "[Desktop Entry]" {
+                    in_main_section = true;
+                    continue;
+                }
+                in_main_section = false;
+            }
+
+            if !in_main_section {
+                continue;
+            }
+
+            // if line.starts_with("TryExec") {
+            //     if let Some(value) = Self::line_value(line) {
+            //         entry.TryExec = Some(value.to_string());
+            //     }
+            // }
```

Comment:
Instead of these if statements, it may look nicer to put them in a match statement. Something like:
```
if let Some((starts_with, value)) = split_line(line) {
    match starts_with {
        "Exec" => entry.exec = Some(value.to_string()),
        "Path" => entry.path = Some(PathBuf::from(value)),
        ...
}

fn split_line(line: &str) -> Option((&str, &str)) {
    line.find('=')?; //Check we have an equals, if we don't return None
    let strings: Vec<&str> = line.split_once('=');
    Some((strings[0], strings[1]))
}
```